letsencrypt-auto fails with SSL error to pypi.python.org

I was told on #letsencrypt to file this as a bug by @hlieberman 

System is Ubuntu Trusty 14.04

root@clients:~/letsencrypt# python --version
Python 2.7.6
root@clients:~/letsencrypt# pip --version
pip 1.5.4 from /usr/lib/python2.7/dist-packages (python 2.7)

Here is the output:

```
root@clients:~/letsencrypt# ./letsencrypt-auto -v --help
Bootstrapping dependencies for Debian-based OSes...
Ign http://mirrors.service.networklayer.com trusty InRelease
Hit http://mirrors.service.networklayer.com trusty-updates InRelease           
Hit http://mirrors.service.networklayer.com trusty-backports InRelease         
Hit http://mirrors.service.networklayer.com trusty-security InRelease          
Hit http://mirrors.service.networklayer.com trusty Release.gpg                 
Hit http://mirrors.service.networklayer.com trusty Release                     
Get:1 http://security.ubuntu.com trusty-security InRelease [65.9 kB]           
Hit http://repo.percona.com trusty InRelease                                   
Hit http://mirrors.service.networklayer.com trusty-updates/main Sources        
Ign http://apt.mailspect.com mpp-release InRelease                             
Hit http://mirrors.service.networklayer.com trusty-updates/restricted Sources  
Hit http://mirrors.service.networklayer.com trusty-updates/universe Sources    
Hit http://mirrors.service.networklayer.com trusty-updates/multiverse Sources  
Hit http://mirrors.service.networklayer.com trusty-updates/main amd64 Packages 
Hit http://mirrors.service.networklayer.com trusty-updates/restricted amd64 Packages
Hit http://mirrors.service.networklayer.com trusty-updates/universe amd64 Packages
Hit http://mirrors.service.networklayer.com trusty-updates/multiverse amd64 Packages
Hit http://mirrors.service.networklayer.com trusty-updates/main i386 Packages  
Hit http://mirrors.service.networklayer.com trusty-updates/restricted i386 Packages
Hit http://mirrors.service.networklayer.com trusty-updates/universe i386 Packages
Hit http://apt.mailspect.com mpp-release Release.gpg                           
Hit http://mirrors.service.networklayer.com trusty-updates/multiverse i386 Packages
Hit http://mirrors.service.networklayer.com trusty-updates/main Translation-en 
Hit http://mirrors.service.networklayer.com trusty-updates/multiverse Translation-en
Hit http://mirrors.service.networklayer.com trusty-updates/restricted Translation-en
Hit http://mirrors.service.networklayer.com trusty-updates/universe Translation-en
Hit http://mirrors.service.networklayer.com trusty-backports/main Sources      
Hit http://mirrors.service.networklayer.com trusty-backports/restricted Sources
Hit http://mirrors.service.networklayer.com trusty-backports/universe Sources  
Hit http://mirrors.service.networklayer.com trusty-backports/multiverse Sources
Hit http://mirrors.service.networklayer.com trusty-backports/main amd64 Packages
Hit http://ppa.launchpad.net trusty InRelease                                  
Hit http://mirrors.service.networklayer.com trusty-backports/restricted amd64 Packages
Hit http://apt.mailspect.com mpp-release Release                               
Hit http://mirrors.service.networklayer.com trusty-backports/universe amd64 Packages
Hit http://mirrors.service.networklayer.com trusty-backports/multiverse amd64 Packages
Hit http://mirrors.service.networklayer.com trusty-backports/main i386 Packages
Hit http://mirrors.service.networklayer.com trusty-backports/restricted i386 Packages
Hit http://mirrors.service.networklayer.com trusty-backports/universe i386 Packages
Hit http://mirrors.service.networklayer.com trusty-backports/multiverse i386 Packages
Hit http://mirrors.service.networklayer.com trusty-backports/main Translation-en
Hit http://mirrors.service.networklayer.com trusty-backports/multiverse Translation-en
Hit http://mirrors.service.networklayer.com trusty-backports/restricted Translation-en
Hit http://mirrors.service.networklayer.com trusty-backports/universe Translation-en
Hit http://mirrors.service.networklayer.com trusty-security/main Sources       
Hit http://mirrors.service.networklayer.com trusty-security/restricted Sources 
Hit http://mirrors.service.networklayer.com trusty-security/universe Sources   
Hit http://repo.percona.com trusty/main Sources                                
Hit http://mirrors.service.networklayer.com trusty-security/multiverse Sources 
Hit http://mirrors.service.networklayer.com trusty/main Sources                
Hit http://mirrors.service.networklayer.com trusty/restricted Sources          
Hit http://mirrors.service.networklayer.com trusty/universe Sources            
Hit http://mirrors.service.networklayer.com trusty/multiverse Sources          
Hit http://mirrors.service.networklayer.com trusty/main amd64 Packages         
Hit http://mirrors.service.networklayer.com trusty/restricted amd64 Packages   
Get:2 http://security.ubuntu.com trusty-security/main amd64 Packages [421 kB]  
Hit http://mirrors.service.networklayer.com trusty/universe amd64 Packages     
Hit http://mirrors.service.networklayer.com trusty/multiverse amd64 Packages   
Hit http://mirrors.service.networklayer.com trusty/main i386 Packages          
Hit http://mirrors.service.networklayer.com trusty/restricted i386 Packages    
Hit http://mirrors.service.networklayer.com trusty/universe i386 Packages      
Hit http://apt.mailspect.com mpp-release/non-free amd64 Packages               
Hit http://mirrors.service.networklayer.com trusty/multiverse i386 Packages    
Hit http://repo.percona.com trusty/main amd64 Packages                         
Hit http://mirrors.service.networklayer.com trusty/main Translation-en         
Hit http://mirrors.service.networklayer.com trusty/multiverse Translation-en   
Hit http://mirrors.service.networklayer.com trusty/restricted Translation-en   
Hit http://mirrors.service.networklayer.com trusty/universe Translation-en     
Hit http://ppa.launchpad.net trusty InRelease                                  
Hit http://apt.mailspect.com mpp-release/non-free i386 Packages                
Hit http://repo.percona.com trusty/main i386 Packages                          
Ign http://extras.ubuntu.com trusty InRelease                                  
Ign http://mirrors.service.networklayer.com trusty/main Translation-en_US      
Ign http://mirrors.service.networklayer.com trusty/multiverse Translation-en_US
Ign http://mirrors.service.networklayer.com trusty/restricted Translation-en_US
Ign http://mirrors.service.networklayer.com trusty/universe Translation-en_US  
Hit http://ppa.launchpad.net trusty/main Sources                               
Hit http://extras.ubuntu.com trusty Release.gpg                                
Ign http://archive.canonical.com trusty InRelease                              
Hit http://ppa.launchpad.net trusty/main amd64 Packages                        
Hit http://extras.ubuntu.com trusty Release                                    
Hit http://archive.canonical.com trusty Release.gpg                            
Hit http://ppa.launchpad.net trusty/main i386 Packages                         
Hit http://extras.ubuntu.com trusty/main Sources                               
Ign http://apt.mailspect.com mpp-release/non-free Translation-en_US            
Get:3 http://security.ubuntu.com trusty-security/restricted amd64 Packages [13.0 kB]
Hit http://archive.canonical.com trusty Release                                
Ign http://apt.mailspect.com mpp-release/non-free Translation-en               
Get:4 http://security.ubuntu.com trusty-security/universe amd64 Packages [124 kB]
Hit http://ppa.launchpad.net trusty/main Translation-en                        
Hit http://archive.canonical.com trusty/partner Sources                        
Hit http://extras.ubuntu.com trusty/main amd64 Packages                        
Ign http://repo.percona.com trusty/main Translation-en_US                      
Ign http://repo.percona.com trusty/main Translation-en                         
Hit http://ppa.launchpad.net trusty/main Sources                               
Hit http://extras.ubuntu.com trusty/main i386 Packages                         
Hit http://archive.canonical.com trusty/partner amd64 Packages                 
Hit http://ppa.launchpad.net trusty/main amd64 Packages                        
Hit http://archive.canonical.com trusty/partner i386 Packages                  
Hit http://ppa.launchpad.net trusty/main i386 Packages                         
Hit http://archive.canonical.com trusty/partner Translation-en                 
Get:5 http://security.ubuntu.com trusty-security/multiverse amd64 Packages [4,990 B]
Hit http://ppa.launchpad.net trusty/main Translation-en                        
Get:6 http://security.ubuntu.com trusty-security/main i386 Packages [393 kB]   
Get:7 http://security.ubuntu.com trusty-security/restricted i386 Packages [12.7 kB]
Get:8 http://security.ubuntu.com trusty-security/universe i386 Packages [124 kB]
Ign http://extras.ubuntu.com trusty/main Translation-en_US                     
Ign http://extras.ubuntu.com trusty/main Translation-en                        
Get:9 http://security.ubuntu.com trusty-security/multiverse i386 Packages [5,164 B]
Hit http://security.ubuntu.com trusty-security/main Translation-en   
Hit http://security.ubuntu.com trusty-security/multiverse Translation-en
Hit http://security.ubuntu.com trusty-security/restricted Translation-en
Hit http://security.ubuntu.com trusty-security/universe Translation-en
Fetched 1,162 kB in 3s (354 kB/s)            
Reading package lists... Done
Reading package lists... Done
Building dependency tree       
Reading state information... Done
dialog is already the newest version.
gcc is already the newest version.
python is already the newest version.
python-dev is already the newest version.
augeas-lenses is already the newest version.
ca-certificates is already the newest version.
libaugeas0 is already the newest version.
libffi-dev is already the newest version.
python-virtualenv is already the newest version.
libssl-dev is already the newest version.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Checking for new version...
Creating virtual environment...
Running virtualenv with interpreter /usr/bin/python2.7
New python executable in /root/.local/share/letsencrypt/bin/python2.7
Also creating executable in /root/.local/share/letsencrypt/bin/python
Installing setuptools, pip...done.
Installing Python packages...
Had a problem while downloading and verifying Python packages:
Downloading argparse==1.4.0 (23K)...
Downloading cffi==1.4.2 (365K)...
Downloading ConfigArgParse==0.10.0 (36K)...
Downloading configobj==5.0.6 (33K)...
Downloading cryptography==1.1.2 (349K)...
Downloading enum34==1.1.2 (46K)...
Downloading funcsigs==0.4 (14K)...
Downloading idna==2.0 (61K)...
Downloading ipaddress==1.0.16 (17K)...
Downloading linecache2==1.0.0 (12K)...
Downloading ndg-httpsclient==0.4.0 (24K)...
Downloading ordereddict==1.1 (2K)...
Downloading parsedatetime==1.5 (50K)...
Downloading pbr==1.8.1 (89K)...
Downloading psutil==3.3.0 (261K)...
Downloading pyasn1==0.1.9 (36K)...
Downloading pycparser==2.14 (223K)...
Downloading pyOpenSSL==0.15.1 (102K)...
Downloading pyRFC3339==1.0 (8K)...
Downloading python-augeas==0.5.0 (90K)...
Downloading python2-pythondialog==3.3.0 (1831K)...
Downloading pytz==2015.7 (476K)...
Downloading requests==2.9.1 (501K)...
Downloading six==1.10.0 (10K)...
Downloading traceback2==1.4.0 (16K)...
Downloading unittest2==1.1.0 (96K)...
Downloading Werkzeug==0.11.3 (305K)...
Downloading zope.component==4.2.2 (546K)...
Downloading zope.event==4.1.0 (476K)...
Downloading zope.interface==4.1.3 (141K)...
Downloading acme==0.4.0 (95K)...
Downloading letsencrypt==0.4.0 (197K)...
Downloading letsencrypt-apache==0.4.0 (93K)...
Downloading mock==1.0.1 (818K)...
Unpacking /tmp/peep-Ojxgfj/argparse-1.4.0-py2.py3-none-any.whl
Installing collected packages: argparse
Successfully installed argparse
Cleaning up...
Unpacking /tmp/peep-dpxjJY/cffi-1.4.2.tar.gz
  Running setup.py (path:/tmp/pip-VOoOut-build/setup.py) egg_info for package from file:///tmp/peep-dpxjJY/cffi-1.4.2.tar.gz
    
Installing collected packages: cffi
  Running setup.py install for cffi
    building '_cffi_backend' extension
    x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DUSE__THREAD -I/usr/include/python2.7 -c c/_cffi_backend.c -o build/temp.linux-x86_64-2.7/c/_cffi_backend.o
    x86_64-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -Wl,-z,relro -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -D_FORTIFY_SOURCE=2 -g -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security build/temp.linux-x86_64-2.7/c/_cffi_backend.o -lffi -o build/lib.linux-x86_64-2.7/_cffi_backend.so
    
Successfully installed cffi
Cleaning up...
Unpacking /tmp/peep-mZUcZM/ConfigArgParse-0.10.0.tar.gz
  Running setup.py (path:/tmp/pip-0bBqLP-build/setup.py) egg_info for package from file:///tmp/peep-mZUcZM/ConfigArgParse-0.10.0.tar.gz
    
Installing collected packages: ConfigArgParse
  Running setup.py install for ConfigArgParse
    
Successfully installed ConfigArgParse
Cleaning up...
Unpacking /tmp/peep-V7hw8_/configobj-5.0.6.tar.gz
  Running setup.py (path:/tmp/pip-2Pnqqj-build/setup.py) egg_info for package from file:///tmp/peep-V7hw8_/configobj-5.0.6.tar.gz
    
Installing collected packages: configobj
  Running setup.py install for configobj
    
Successfully installed configobj
Cleaning up...
Unpacking /tmp/peep-fP1wgM/cryptography-1.1.2.tar.gz
  Running setup.py (path:/tmp/pip-hzzpRo-build/setup.py) egg_info for package from file:///tmp/peep-fP1wgM/cryptography-1.1.2.tar.gz
    
    no previously-included directories found matching 'docs/_build'
    warning: no previously-included files matching '*' found under directory 'vectors'
Installing collected packages: cryptography
  Running setup.py install for cryptography
    Download error on https://pypi.python.org/simple/pycparser/: [Errno 1] _ssl.c:510: error:14090086:SSL routines:SSL3_GET_SERVER_CERTIFICATE:certificate verify failed -- Some packages may not be found!
    Couldn't find index page for 'pycparser' (maybe misspelled?)
    Download error on https://pypi.python.org/simple/: [Errno 1] _ssl.c:510: error:14090086:SSL routines:SSL3_GET_SERVER_CERTIFICATE:certificate verify failed -- Some packages may not be found!
    No local packages or download links found for pycparser
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-hzzpRo-build/setup.py", line 318, in <module>
        **keywords_with_side_effects(sys.argv)
      File "/usr/lib/python2.7/distutils/core.py", line 111, in setup
        _setup_distribution = dist = klass(attrs)
      File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/setuptools/dist.py", line 239, in __init__
        self.fetch_build_eggs(attrs.pop('setup_requires'))
      File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/setuptools/dist.py", line 264, in fetch_build_eggs
        replace_conflicting=True
      File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/pkg_resources.py", line 580, in resolve
        dist = best[req.key] = env.best_match(req, ws, installer)
      File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/pkg_resources.py", line 818, in best_match
        return self.obtain(req, installer) # try and download/install
      File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/pkg_resources.py", line 830, in obtain
        return installer(requirement)
      File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/setuptools/dist.py", line 314, in fetch_build_egg
        return cmd.easy_install(req)
      File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 587, in easy_install
        raise DistutilsError(msg)
    distutils.errors.DistutilsError: Could not find suitable distribution for Requirement.parse('pycparser')
    Complete output from command /root/.local/share/letsencrypt/bin/python -c "import setuptools, tokenize;__file__='/tmp/pip-hzzpRo-build/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('
', '
'), __file__, 'exec'))" install --record /tmp/pip-3tii3k-record/install-record.txt --single-version-externally-managed --compile --install-headers /root/.local/share/letsencrypt/include/site/python2.7:
    Download error on https://pypi.python.org/simple/pycparser/: [Errno 1] _ssl.c:510: error:14090086:SSL routines:SSL3_GET_SERVER_CERTIFICATE:certificate verify failed -- Some packages may not be found!

Couldn't find index page for 'pycparser' (maybe misspelled?)

Download error on https://pypi.python.org/simple/: [Errno 1] _ssl.c:510: error:14090086:SSL routines:SSL3_GET_SERVER_CERTIFICATE:certificate verify failed -- Some packages may not be found!

No local packages or download links found for pycparser

Traceback (most recent call last):

  File "<string>", line 1, in <module>

  File "/tmp/pip-hzzpRo-build/setup.py", line 318, in <module>

    **keywords_with_side_effects(sys.argv)

  File "/usr/lib/python2.7/distutils/core.py", line 111, in setup

    _setup_distribution = dist = klass(attrs)

  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/setuptools/dist.py", line 239, in __init__

    self.fetch_build_eggs(attrs.pop('setup_requires'))

  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/setuptools/dist.py", line 264, in fetch_build_eggs

    replace_conflicting=True

  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/pkg_resources.py", line 580, in resolve

    dist = best[req.key] = env.best_match(req, ws, installer)

  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/pkg_resources.py", line 818, in best_match

    return self.obtain(req, installer) # try and download/install

  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/pkg_resources.py", line 830, in obtain

    return installer(requirement)

  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/setuptools/dist.py", line 314, in fetch_build_egg

    return cmd.easy_install(req)

  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 587, in easy_install

    raise DistutilsError(msg)

distutils.errors.DistutilsError: Could not find suitable distribution for Requirement.parse('pycparser')

----------------------------------------
Cleaning up...
Command /root/.local/share/letsencrypt/bin/python -c "import setuptools, tokenize;__file__='/tmp/pip-hzzpRo-build/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('
', '
'), __file__, 'exec'))" install --record /tmp/pip-3tii3k-record/install-record.txt --single-version-externally-managed --compile --install-headers /root/.local/share/letsencrypt/include/site/python2.7 failed with error code 1 in /tmp/pip-hzzpRo-build
Storing debug log for failure in /root/.pip/pip.log
```

The certificate bundle for the system seems fine, I can connect using curl or openssl:

```
root@clients:~/letsencrypt# openssl s_client -connect pypi.python.org:443 -showcerts
CONNECTED(00000003)
depth=2 C = US, O = DigiCert Inc, OU = www.digicert.com, CN = DigiCert High Assurance EV Root CA
verify return:1
depth=1 C = US, O = DigiCert Inc, OU = www.digicert.com, CN = DigiCert SHA2 Extended Validation Server CA
verify return:1
depth=0 businessCategory = Private Organization, jurisdictionC = US, jurisdictionST = Delaware, serialNumber = 3359300, street = 16 Allen Rd, postalCode = 03894-4801, C = US, ST = NH, L = "Wolfeboro,", O = Python Software Foundation, CN = www.python.org
verify return:1
---
Certificate chain
 0 s:/businessCategory=Private Organization/jurisdictionC=US/jurisdictionST=Delaware/serialNumber=3359300/street=16 Allen Rd/postalCode=03894-4801/C=US/ST=NH/L=Wolfeboro,/O=Python Software Foundation/CN=www.python.org
   i:/C=US/O=DigiCert Inc/OU=www.digicert.com/CN=DigiCert SHA2 Extended Validation Server CA
-----BEGIN CERTIFICATE-----
MIIIXDCCB0SgAwIBAgIQC1o1AAqFW/41glwVSXUq/zANBgkqhkiG9w0BAQsFADB1
MQswCQYDVQQGEwJVUzEVMBMGA1UEChMMRGlnaUNlcnQgSW5jMRkwFwYDVQQLExB3
d3cuZGlnaWNlcnQuY29tMTQwMgYDVQQDEytEaWdpQ2VydCBTSEEyIEV4dGVuZGVk
IFZhbGlkYXRpb24gU2VydmVyIENBMB4XDTE0MDkwNTAwMDAwMFoXDTE2MDkwOTEy
MDAwMFowgfkxHTAbBgNVBA8TFFByaXZhdGUgT3JnYW5pemF0aW9uMRMwEQYLKwYB
BAGCNzwCAQMTAlVTMRkwFwYLKwYBBAGCNzwCAQITCERlbGF3YXJlMRAwDgYDVQQF
EwczMzU5MzAwMRQwEgYDVQQJEwsxNiBBbGxlbiBSZDETMBEGA1UEERMKMDM4OTQt
NDgwMTELMAkGA1UEBhMCVVMxCzAJBgNVBAgTAk5IMRMwEQYDVQQHEwpXb2xmZWJv
cm8sMSMwIQYDVQQKExpQeXRob24gU29mdHdhcmUgRm91bmRhdGlvbjEXMBUGA1UE
AxMOd3d3LnB5dGhvbi5vcmcwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIB
AQCtUnfHpOteoIqZxGsaR/2tIenj0+pBtNBiWT6PlYLLXC6MNRjFwtnhRzEVanAm
GEEOEQwUokYZHw8kCL2SIZ1DFI5IIFyhTFql1dqiKtoQse0LAZlUHscVxn9OZyWM
DA4JZ6A4c3/j5SA9hGO3+KyTc95GfiEXqkSkmjH3aBtY2flr+H1fvatQA8AIAD5k
weQLFbbqi33Uvf4sJ3OhY63Kf1ZWteXSeCT+FRMlFTaYbauo86AmU9X2/b85wold
naUO3VjcGjTSoSuaxtWuHFRxpOTBG7bqPbtWk+X5l+rjsIoGJ6ZrRFbAtHqG+S3v
luEG9FtgGAo+3hKm99U8UKKVAgMBAAGjggRhMIIEXTAfBgNVHSMEGDAWgBQ901Cl
1qCt7vNKYApl0yHU+PjWDzAdBgNVHQ4EFgQUTWfmKThuIBhkZX4B3yNf+DpBqokw
ggEUBgNVHREEggELMIIBB4IOd3d3LnB5dGhvbi5vcmeCCnB5dGhvbi5vcmeCD3B5
cGkucHl0aG9uLm9yZ4IPZG9jcy5weXRob24ub3JnghN0ZXN0cHlwaS5weXRob24u
b3Jngg9idWdzLnB5dGhvbi5vcmeCD3dpa2kucHl0aG9uLm9yZ4INaGcucHl0aG9u
Lm9yZ4IPbWFpbC5weXRob24ub3JnghRwYWNrYWdpbmcucHl0aG9uLm9yZ4IQcHl0
aG9uaG9zdGVkLm9yZ4IUd3d3LnB5dGhvbmhvc3RlZC5vcmeCFXRlc3QucHl0aG9u
aG9zdGVkLm9yZ4IMdXMucHljb24ub3Jngg1pZC5weXRob24ub3JnMA4GA1UdDwEB
/wQEAwIFoDAdBgNVHSUEFjAUBggrBgEFBQcDAQYIKwYBBQUHAwIwdQYDVR0fBG4w
bDA0oDKgMIYuaHR0cDovL2NybDMuZGlnaWNlcnQuY29tL3NoYTItZXYtc2VydmVy
LWcxLmNybDA0oDKgMIYuaHR0cDovL2NybDQuZGlnaWNlcnQuY29tL3NoYTItZXYt
c2VydmVyLWcxLmNybDBCBgNVHSAEOzA5MDcGCWCGSAGG/WwCATAqMCgGCCsGAQUF
BwIBFhxodHRwczovL3d3dy5kaWdpY2VydC5jb20vQ1BTMIGIBggrBgEFBQcBAQR8
MHowJAYIKwYBBQUHMAGGGGh0dHA6Ly9vY3NwLmRpZ2ljZXJ0LmNvbTBSBggrBgEF
BQcwAoZGaHR0cDovL2NhY2VydHMuZGlnaWNlcnQuY29tL0RpZ2lDZXJ0U0hBMkV4
dGVuZGVkVmFsaWRhdGlvblNlcnZlckNBLmNydDAMBgNVHRMBAf8EAjAAMIIBfgYK
KwYBBAHWeQIEAgSCAW4EggFqAWgAdQCkuQmQtBhYFIe7E6LMZ3AKPDWYBPkb37jj
d80OyA3cEAAAAU21GPxpAAAEAwBGMEQCIGLF4WOctBDAHh+b90vonuTLiIUUwk/Q
phrYWYKF5mq3AiBfiUrG1PQtEZPNLyGU1OQtaj+FNhwHQofitBZkYZ0+cQB3AGj2
mPgfZIK+OozuuSgdTPxxUV1nk9RE0QpnrLtPT/vEAAABTbUY/FEAAAQDAEgwRgIh
AMSX5QOeFcssSUnVyWQrgLnSYw37CTFNXGnuZuF3/4K4AiEAg1MqScP+dMT6wl81
I1PumQ4v76Azpn2YLb2NQ1SWsfEAdgBWFAaaL9fC7NP14b1Esj7HRna5vJkRXMDv
lJhV1onQ3QAAAU21GP2YAAAEAwBHMEUCIQCYsROFukGKYaQvnS3pO+IavIlQI17F
Uj24bobC9r8RugIgLNIcL+5Dbedz9bL4xHVl/aAIj/Wt1dmMsq9noUQW+FgwDQYJ
KoZIhvcNAQELBQADggEBAEZfzjVPEVWR0wXyZ9GVKD/7KvWRErcCQI6oadFUu7nj
IAmkyoUFFUhW+hFCrGmf3FO3fSo00HG8fhQDjObPEqThpNKc0qdKbKNnvCFXhoQG
hVzRXOAZSSmb6fOalhHx5ZFyjksbohsfDDX+XTo9dYAfkEDKFMAAoRS/FXNdIm8O
1ejT8pbPrGQjFI7q6I6EnhrMfLfHo9XQv0gRxF7CbHFy0ImnjnN4MhWtoo/w4kMb
fq9CC5tdzxj9hFPvDKdG8eUvsGhva8cREwwgpBbjGYPhnVAkoRAnwFNEeRcPU/QZ
DoQj1GhWTwdVrIYKr70tfuhQ+ZzL3XQo+VEoYhwbKRM=
-----END CERTIFICATE-----
 1 s:/C=US/O=DigiCert Inc/OU=www.digicert.com/CN=DigiCert SHA2 Extended Validation Server CA
   i:/C=US/O=DigiCert Inc/OU=www.digicert.com/CN=DigiCert High Assurance EV Root CA
-----BEGIN CERTIFICATE-----
MIIEtjCCA56gAwIBAgIQDHmpRLCMEZUgkmFf4msdgzANBgkqhkiG9w0BAQsFADBs
MQswCQYDVQQGEwJVUzEVMBMGA1UEChMMRGlnaUNlcnQgSW5jMRkwFwYDVQQLExB3
d3cuZGlnaWNlcnQuY29tMSswKQYDVQQDEyJEaWdpQ2VydCBIaWdoIEFzc3VyYW5j
ZSBFViBSb290IENBMB4XDTEzMTAyMjEyMDAwMFoXDTI4MTAyMjEyMDAwMFowdTEL
MAkGA1UEBhMCVVMxFTATBgNVBAoTDERpZ2lDZXJ0IEluYzEZMBcGA1UECxMQd3d3
LmRpZ2ljZXJ0LmNvbTE0MDIGA1UEAxMrRGlnaUNlcnQgU0hBMiBFeHRlbmRlZCBW
YWxpZGF0aW9uIFNlcnZlciBDQTCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoC
ggEBANdTpARR+JmmFkhLZyeqk0nQOe0MsLAAh/FnKIaFjI5j2ryxQDji0/XspQUY
uD0+xZkXMuwYjPrxDKZkIYXLBxA0sFKIKx9om9KxjxKws9LniB8f7zh3VFNfgHk/
LhqqqB5LKw2rt2O5Nbd9FLxZS99RStKh4gzikIKHaq7q12TWmFXo/a8aUGxUvBHy
/Urynbt/DvTVvo4WiRJV2MBxNO723C3sxIclho3YIeSwTQyJ3DkmF93215SF2AQh
cJ1vb/9cuhnhRctWVyh+HA1BV6q3uCe7seT6Ku8hI3UarS2bhjWMnHe1c63YlC3k
8wyd7sFOYn4XwHGeLN7x+RAoGTMCAwEAAaOCAUkwggFFMBIGA1UdEwEB/wQIMAYB
Af8CAQAwDgYDVR0PAQH/BAQDAgGGMB0GA1UdJQQWMBQGCCsGAQUFBwMBBggrBgEF
BQcDAjA0BggrBgEFBQcBAQQoMCYwJAYIKwYBBQUHMAGGGGh0dHA6Ly9vY3NwLmRp
Z2ljZXJ0LmNvbTBLBgNVHR8ERDBCMECgPqA8hjpodHRwOi8vY3JsNC5kaWdpY2Vy
dC5jb20vRGlnaUNlcnRIaWdoQXNzdXJhbmNlRVZSb290Q0EuY3JsMD0GA1UdIAQ2
MDQwMgYEVR0gADAqMCgGCCsGAQUFBwIBFhxodHRwczovL3d3dy5kaWdpY2VydC5j
b20vQ1BTMB0GA1UdDgQWBBQ901Cl1qCt7vNKYApl0yHU+PjWDzAfBgNVHSMEGDAW
gBSxPsNpA/i/RwHUmCYaCALvY2QrwzANBgkqhkiG9w0BAQsFAAOCAQEAnbbQkIbh
hgLtxaDwNBx0wY12zIYKqPBKikLWP8ipTa18CK3mtlC4ohpNiAexKSHc59rGPCHg
4xFJcKx6HQGkyhE6V6t9VypAdP3THYUYUN9XR3WhfVUgLkc3UHKMf4Ib0mKPLQNa
2sPIoc4sUqIAY+tzunHISScjl2SFnjgOrWNoPLpSgVh5oywM395t6zHyuqB8bPEs
1OG9d4Q3A84ytciagRpKkk47RpqF/oOi+Z6Mo8wNXrM9zwR4jxQUezKcxwCmXMS1
oVWNWlZopCJwqjyBcdmdqEU79OX2olHdx3ti6G8MdOu42vi/hw15UJGQmxg7kVkn
8TUoE6smftX3eg==
-----END CERTIFICATE-----
---
Server certificate
subject=/businessCategory=Private Organization/jurisdictionC=US/jurisdictionST=Delaware/serialNumber=3359300/street=16 Allen Rd/postalCode=03894-4801/C=US/ST=NH/L=Wolfeboro,/O=Python Software Foundation/CN=www.python.org
issuer=/C=US/O=DigiCert Inc/OU=www.digicert.com/CN=DigiCert SHA2 Extended Validation Server CA
---
No client certificate CA names sent
Peer signing digest: SHA512
Server Temp Key: ECDH, P-256, 256 bits
---
SSL handshake has read 4016 bytes and written 441 bytes
---
New, TLSv1/SSLv3, Cipher is ECDHE-RSA-AES128-GCM-SHA256
Server public key is 2048 bit
Secure Renegotiation IS supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : ECDHE-RSA-AES128-GCM-SHA256
    Session-ID: 7F3B51B60EC88FDCC365097FEE5C82A6672AEC0FAE05825F2C9CE7748E34E11A
    Session-ID-ctx: 
    Master-Key: 5A2B4831AC1AEDFC54E79F79AB1862F1098B716B8B6010F5961E45C8879D7118A8F909326ADD9BF5A355CDE357FB1C78
    Key-Arg   : None
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    TLS session ticket lifetime hint: 1200 (seconds)
    TLS session ticket:
    0000 - 63 cc 77 4a 00 db 2c 42-2e 8f 76 23 dd a9 ae 53   c.wJ..,B..v#...S
    0010 - 23 11 7f 5d 9d e3 70 1e-4d a6 a3 17 a3 42 e9 8e   #..]..p.M....B..
    0020 - c1 21 f7 39 98 7b fe e2-62 05 fc 32 59 92 51 be   .!.9.{..b..2Y.Q.
    0030 - a4 e6 df 45 7b be 01 1d-13 1c e1 ec 93 e0 bf 9c   ...E{...........
    0040 - db ba 76 54 e5 c2 f0 95-1d 55 d4 60 c8 ee 2e e1   ..vT.....U.`....
    0050 - 76 57 96 48 05 6d b5 d1-88 5e 73 a2 c1 8e d7 b2   vW.H.m...^s.....
    0060 - 7f 5d 38 55 98 4a 9e 92-61 63 38 f4 08 0a 58 37   .]8U.J..ac8...X7
    0070 - 99 ac 67 4b 4f 3a 19 71-de 44 e3 70 dd b7 1a 37   ..gKO:.q.D.p...7
    0080 - 6c 65 0f e3 f5 e9 7e d5-38 f7 08 9d f4 ef e0 c5   le....~.8.......
    0090 - 7a 67 73 da 5d 18 0f f0-cd 4b 50 e3 7a 4b 9d 7f   zgs.]....KP.zK..

    Start Time: 1455772185
    Timeout   : 300 (sec)
    Verify return code: 0 (ok)
---
```

--- a/letsencrypt-auto-source/letsencrypt-auto
+++ b/letsencrypt-auto-source/letsencrypt-auto
@@ -455,6 +455,11 @@ if [ "$1" = "--le-auto-phase2" ]; then
 # sha256: YrCJpVvh2JSc0rx-DfC9254Cj678jDIDjMhIYq791uQ
 argparse==1.4.0
 
+# This comes before cffi because cffi will otherwise install an unchecked
+# version via setup_requires.
+# sha256: eVm0p0q9wnsxL-0cIebK-TCc4LKeqGtZH9Lpns3yf3M
+pycparser==2.14
+
 # sha256: U8HJ3bMEMVE-t_PN7wo-BrDxJSGIqqd0SvD1pM1F268
 # sha256: pWj0nfyhKo2fNwGHJX78WKOBCeHu5xTZKFYdegGKZPg
 # sha256: gJxsqM-8ruv71DK0V2ABtA04_yRjdzy1dXfXXhoCC8M
@@ -571,9 +576,6 @@ psutil==3.3.0
 # sha256: gyPgNjey0HLMcEEwC6xuxEjDwolQq0A3YDZ4jpoa9ik
 # sha256: hTys2W0fcB3dZ6oD7MBfUYkBNbcmLpInEBEvEqLtKn8
 pyasn1==0.1.9
-
-# sha256: eVm0p0q9wnsxL-0cIebK-TCc4LKeqGtZH9Lpns3yf3M
-pycparser==2.14
 
 # sha256: iORea7Jd_tJyoe8ucoRh1EtjTCzWiemJtuVqNJxaOuU
 # sha256: 8KJgcNbbCIHei8x4RpNLfDyTDY-cedRYg-5ImEvA1nI

--- a/letsencrypt-auto-source/pieces/letsencrypt-auto-requirements.txt
+++ b/letsencrypt-auto-source/pieces/letsencrypt-auto-requirements.txt
@@ -6,6 +6,11 @@
 # sha256: YrCJpVvh2JSc0rx-DfC9254Cj678jDIDjMhIYq791uQ
 argparse==1.4.0
 
+# This comes before cffi because cffi will otherwise install an unchecked
+# version via setup_requires.
+# sha256: eVm0p0q9wnsxL-0cIebK-TCc4LKeqGtZH9Lpns3yf3M
+pycparser==2.14
+
 # sha256: U8HJ3bMEMVE-t_PN7wo-BrDxJSGIqqd0SvD1pM1F268
 # sha256: pWj0nfyhKo2fNwGHJX78WKOBCeHu5xTZKFYdegGKZPg
 # sha256: gJxsqM-8ruv71DK0V2ABtA04_yRjdzy1dXfXXhoCC8M
@@ -122,9 +127,6 @@ psutil==3.3.0
 # sha256: gyPgNjey0HLMcEEwC6xuxEjDwolQq0A3YDZ4jpoa9ik
 # sha256: hTys2W0fcB3dZ6oD7MBfUYkBNbcmLpInEBEvEqLtKn8
 pyasn1==0.1.9
-
-# sha256: eVm0p0q9wnsxL-0cIebK-TCc4LKeqGtZH9Lpns3yf3M
-pycparser==2.14
 
 # sha256: iORea7Jd_tJyoe8ucoRh1EtjTCzWiemJtuVqNJxaOuU
 # sha256: 8KJgcNbbCIHei8x4RpNLfDyTDY-cedRYg-5ImEvA1nI

